lib/ExpensiMark.js

```diff
@@ -402,12 +402,18 @@ export default class ExpensiMark {
             if (abort) {
                 replacedText = replacedText.concat(textToCheck.substr(match.index, (match[0].length)));
             } else {
-                // `match[1]` contains the text inside []. Text is already escaped due to the rules
-                // that precede the link rule (eg, codeFence, inlineCodeBlock, email).
-                const linkText = this.replace(match[1], {
-                    filterRules: ['bold', 'strikethrough', 'italic'],
-                    shouldEscapeText: false,
-                });
+                let linkText = match[1];
+
+                // We don't want to apply filterRules if the linkText is entirely a link
```

Comment:
I think this comment would be more valuable if it explained **why** we don't want to "apply `filterRules`" and what that means exactly. It's also a little unclear what "entirely a link" means vs. something that is "partially a link" or some other meaning.